client/client.py

```diff
@@ -0,0 +1,126 @@
+from datetime import datetime
+from dataclasses import dataclass
+
+
+@dataclass
+class Status:
+    """Represents the status of a file upload and processing."""
+    status: str
+    filename: str
+    timestamp: datetime
+    explanation: str
+
+    def is_done(self) -> bool:
+        """Check if the status is 'done'."""
+        return self.status == 'done'
+
+
+class WebAppClient:
+    """Client for interacting with the web application."""
+
+    def __init__(self, base_url: str):
+        self.base_url = base_url
+
+    def upload(self, file_path: str) -> str:
+        """
+        Upload a file to the web application.
+        Args:
+            file_path (str): The path to the file to upload.
+        Returns:
+            str: The unique identifier (UID) associated with the uploaded file.
+        Raises:
+            Exception: If the upload fails with a non-200 status code.
+        """
+        url = f"{self.base_url}/upload"
+        with open(file_path, 'rb') as file:
+            response = requests.post(url, files={'file': file})
+
+        if response.status_code == 200:
+            json_data = response.json()
+            return json_data['uid']
+        else:
+            raise Exception(f"Upload failed with status code {response.status_code}")
+
+    def status(self, uid: str) -> type[Status]:
+        """
+        Retrieve the status of a file upload.
+        Args:
+            uid (str): The unique identifier (UID) associated with the file.
+        Returns:
+            Status: The status of the file upload.
+        Raises:
+            Exception: If the status retrieval fails with a non-200 status code.
+        """
+        url = f"{self.base_url}/status/{uid}"
+        params = uid
+        response = requests.get(url, params=params)
+
+        if response.status_code == requests.codes.ok:
+            json_data = response.json()
+            status = json_data['status']
+            filename = json_data['filename']
+            timestamp = datetime.strptime(json_data['timestamp'], '%Y-%m-%d %H:%M:%S')
+            explanation = json_data['explanation']
+            return Status(status, filename, timestamp, explanation)
+        else:
+            raise Exception(f"Status retrieval failed with status code {response.status_code}")
+
+    def run_client(self) -> None:
+        """Run the WebAppClient and interact with the web application."""
+        print("Welcome to the Web App Client!")
+        while True:
+            print("\nSelect an option:")
+            print("1. Upload file")
+            print("2. Check status")
+            print("0. Exit")
+            choice = input("Enter your choice: ")
+
+            if choice == '1':
+                file_path = WebAppClient.select_file()
+                if file_path:
+                    try:
+                        uid = self.upload(file_path)
+                        print(f"File uploaded successfully. UID: {uid}")
+                    except Exception as e:
```

Comment:
This name does not adhere to python's naming conventions. Names should be indicative, and when used properly can reduce redundant documentation. In addition, each Variable, function, module, constant and class names has a specific naming convention. Read this link to learn the proper conventions - https://visualgit.readthedocs.io/en/latest/pages/naming_convention.html. If you don't know what CamelCase and snake_case means, read this link - https://www.google.com/amp/s/www.theserverside.com/answer/Camel-case-vs-snake-case-Whats-the-difference%3famp=1 . For a general overview on python conventions (PEP-8), read this link - https://peps.python.org/pep-0008